manager/assets/modext/widgets/media/modx.browser.js

```diff
@@ -189,25 +189,11 @@ Ext.extend(MODx.browser.View,MODx.DataView,{
         w.show(e.target);
     }
 
-    ,downloadFile: function(item,e) {
-        var node = this.cm.activeNode;
-        var data = this.lookup[node.id];
-        MODx.Ajax.request({
-            url: MODx.config.connector_url
-            ,params: {
-                action: 'Browser/File/Download'
-                ,file: data.pathRelative
-                ,wctx: MODx.ctx || ''
-                ,source: this.config.source
-            }
-            ,listeners: {
-                'success':{fn:function(r) {
-                    if (!Ext.isEmpty(r.object.url)) {
-                        location.href = MODx.config.connector_url+'?action=Browser/File/Download&download=1&file='+r.object.url+'&HTTP_MODAUTH='+MODx.siteId+'&source='+this.config.source+'&wctx='+MODx.ctx;
```

Comment:
It's not a degression, it simply doesn't work the way it is written. You have to take into account what's happening in the processor as well to see why.

Reply:
The patch is a degression in terms of good error handling. The code previously checked for the possibility of a download and this is now dropped. The functionality is back at the expense of good error handling. This is the case here and that's the reason for the code review process.

A possible valid fix would be to send `data.pathRelative` in the `file` property in the second request instead of `r.object.url` in the original code. But I did not test that.

Reply:
The return value from the first request `r.object.url` is **not** empty, even if the file doesn't exist anymore.

Reply:
But then the success listener is not triggered.

Reply:
When I open the Media Browser, then delete an image file on the file system and then execute right-click -> "Download File" on the deleted file, the first AJAX request is successful and `object.url` filled.
```
{"success":true,"message":"","total":0,"data":[],"object":{"url":"%2Fmodx300%2Fmyimage.jpg"}}
```
The success listener is triggered.

Reply:
😳

Reply:
I have prepared a check, but my laptop currently has a composer issue. So I can't update the core from 3.0.0-beta to 3.0.0 to test this: https://github.com/Jako/revolution/tree/fix-download-exist

Reply:
@smg6511 Please do not think that your code change was not correct, but it somehow went in the wrong direction. I first had to understand why the previous code ran with two requests.

From the current point of view, the additional FileDownload Helper method suggested at the beginning of this thread is no longer strictly necessary. But it provides the only way to tell the user that the download has been completed successfully. This could be displayed with an info bubble on the right side of the browser viewport.

Reply:
@Jako No worries. I come at this knowing from previous experience with trying to fix the static elements issue, that delving into a file handling/media source-centered issue _could_ become a longer than expected headache. So I was going with more immediacy toward just getting the download working when it _should_, leaving the intended error handling (which was not clearly self-evident from the original code) to a date soon after an initial fix.

At any rate, thankfully you identified a simple solution in #16152 (which I just reviewed). I had just a couple small suggestions there.

I think the push notification-like feedback indicating a successful download is interesting (icing on the cake). You could look to replicate the same feedback mechanism/style as is shown when saving Resources and Elements.